### PR TITLE
feat: upsert search attributes

### PR DESCRIPTION
**Pull request by cv65kr**
# Reason for This PR

https://github.com/temporalio/sdk-php/issues/76

Same functionality like in GO-SDK - https://github.com/temporalio/samples-go/tree/main/searchattributes

Changes needed in PHP-SDK - https://github.com/temporalio/sdk-php/pull/248

## Description of Changes

Added missing implementation of `UpsertWorkflowSearchAttributes` command.

## E2E
Changes: https://github.com/roadrunner-server/rr-e2e-tests/pull/138

Workflow file: https://github.com/temporalio/sdk-php/blob/6a40e81ba7125ee32fbb242599f65ddff649500a/tests/Fixtures/src/Workflow/UpsertSearchAttributesWorkflow.php

Also definition of search attributes needs to be executed to test server before running workflow - https://github.com/temporalio/sdk-php/blob/6a40e81ba7125ee32fbb242599f65ddff649500a/tests/SearchAttributeTestInvoker.php

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the MIT license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

**Comment by rustatian** [comment]
Hey @cv65kr 👋🏻 
Please, do not edit the PR template.

File: aggregatedpool/handler.go [hunk 1]
```diff
@@ -187,6 +187,14 @@ func (wp *Workflow) handleMessage(msg *internal.Message) error {
 			WorkflowTaskTimeout:      command.Options.WorkflowTaskTimeout,
 		})
 
+	case *internal.UpsertWorkflowSearchAttributes:
+		searchAttributes := make(map[string]interface{})
+		err := wp.env.GetDataConverter().FromPayloads(msg.Payloads, &searchAttributes)
+		if err != nil {
+			return errors.E(op, err)
+		}
+		wp.env.UpsertSearchAttributes(searchAttributes)
```
Comment (rustatian) [inline aggregatedpool/handler.go:196]: `UpsertSearchAttributes` may return an error; please, check it.

File: internal/protocol.go [hunk 2]
```diff
@@ -230,6 +231,11 @@ type ContinueAsNew struct {
 	} `json:"options"`
 }
 
+// UpsertWorkflowSearchAttributes allows to upsert search attributes
+type UpsertWorkflowSearchAttributes struct {
+	SearchAttributes map[string]interface{} `json:"searchAttributes"`
```
Comment (rustatian) [inline internal/protocol.go:236]:
```suggestion
	SearchAttributes map[string]any `json:"searchAttributes"`
```

**Comment by cv65kr** [comment]
@rustatian Updated, also signed my commits. And big sorry for removing template 😄

**Comment by cv65kr** [comment]
@rustatian fixed and tested. I did small mistake and encoder was not needed.

image

**Comment by rustatian** [comment]
@cv65kr Could you please add a PHP worker to the description (which sends this command)? I'll add it to our international tests: https://github.com/roadrunner-server/rr-e2e-tests.

**Comment by cv65kr** [comment]
@rustatian You meant workflow definition? I see general worker https://github.com/roadrunner-server/rr-e2e-tests/blob/master/php_test_files/temporal-worker.php

Also added info about definition of search attributes which need to be executed before running such test.

**Comment by cv65kr** [comment]
@rustatian to make your live easier I prepared changes in E2E repository - https://github.com/roadrunner-server/rr-e2e-tests/pull/138 Feel free to modify it since I put my branches there, because we need to wait for merge.

**Comment by rustatian** [comment]
Great, thank you, @cv65kr 👍🏻 
I'll review and merge it tomorrow ⚡

**Review by rustatian — APPROVED** [review]
lgtm

**Pull request merged**